content/engineering/implementing-oidc-sso-having-login-radius-as-identity-provider/index.md

```diff
@@ -0,0 +1,316 @@
+
+
+After setting up the oidc app from the login radius dashboard we will use the go -oidc library to further configure our provider and configure the oidc connect.
+
+## Setting up OIDC Application
+
+First go to Admin Dashboard and Navigate to 
+
+Platform Configuration > Access Configuration > Federated SSO > OpenID Connect
+
+Click on **Add App button**
+
+![OIDC App Configuration](OIDC-App.jpeg)
+
+Enter the **App name** and click one of the following
+
+**Native App**, **Single page App** or **Web App** according to your application
+
+Once clicked on any one of the options u will get 
+
+![OIDC App Setup](App-Setup.jpeg)
+
+#### Upon reaching the configuration page for your OIDC Application, you'll encounter a variety of fields ripe for customization:
+
+1) **Algorithm**: Presently, we offer support for **rs256**.
+2) **Grant Type**: Options include authorization code, implicit, password creds, etc.
+3) You can tailor settings for **Token Expiry**, **Refresh Token**, and **TTL** to suit your needs.
+4) **Data Mapping**: Define fields or properties to be included in the data response.
+5) **Metadata**: Incorporate static, non-profile values into the data response.
+6) Define the Scope for Management API.
+   
+
+This array of configurable options empowers you to fine-tune your OIDC Application according to your specific requirements.
+
+
+## Whitelisting Domain of your Application
+
+To ensure seamless redirection of requests and successful callbacks to your endpoint, we need to add your application's domain to the whitelist. This will authorize the redirection process and prevent any failures when calling the callback endpoint.
+ 
+#### To access Web Apps in Deployment, follow these steps:
+
+1) Navigate to the Deployment section from the Dashboard.
+2) Once in Deployment, select the Apps tab.
+3) From there, choose Web Apps.
+
+#### Now, to add a new site:
+
+1) Click on the Add New Site button.
+2) Enter the domain name of the website (example : "https://localhost:8080").
+
+![Whitelisting Domain Name](Whitelisting-Domain.jpeg)
+
+
+### Whitelisting Domain from OIDC Application configuration.
+Login Radius gives the feature to uniquely identify the redirect urls for individual OIDC applications.
+
+
+* When setting the configuration of the OIDC Application you can specify the redirect url of your backend
+and it will be whitelisted .
+* The field name is **Login Redirect URL**.
+
+
+
+## Setting up the provider object and the OAuthconfig with the loginradius OIDC App  credentials
+
+```javascript
+provider, err := oidc.NewProvider(ctx, "https://api.loginradius.com/{oidcappname}")
+if err != nil {
+    // handle error
+}
+
+// Configure an OpenID Connect aware OAuth2 client.
+oauth2Config := oauth2.Config{
+    ClientID:     your-oidc-clientID,
+    ClientSecret: your-oidc-clientSecret
+    RedirectURL:  redirectURL,
```

Comment:
I think you should cover the details of how to get this client ID and Client's secret